DAQ_System/Source/main.cpp

```diff
@@ -46,154 +51,63 @@ int main() {
     // File system declaration
     FATFileSystem   file_system("fs");
     
-    printf("--- Mbed OS filesystem example ---\n");
+    // buffer string used to write to a file
+    char buffer[kBlockSectorByteSize] = "\0"; 
+
+    // used for error checking when operating the block device
+    int status = 0;
+
+    // sensor values
+    int linpot1 = 0;
+    int linpot2 = 0;
+    int linpot3 = 0;
+    int linpot4 = 0;
+
+    printf("--- CSV File Test ---\n");
 
     // Try to mount the filesystem
     printf("Mounting the filesystem... ");
     fflush(stdout);
     Mount(&file_system, &block_device);
 
-    int status = 0;
     // Open the numbers file
-    printf("Opening \"/fs/numbers.txt\"... ");
+    printf("Opening \"/fs/data.csv\"... ");
     fflush(stdout);
 
-    FILE* numbers_file = FileOpen("/fs/numbers.txt");
-
-    char buffer[32] = "\0"; // buffer string used to write to a file
-
-    for (int i = 0; i < 10; i++) {
-        printf("\rWriting numbers (%d/%d)... ", i, 10);
-        fflush(stdout);
-        sprintf(buffer, "   %d\n", i); // we use this to print our desired format to the buffer string which is then written to the file
-        FileWrite(numbers_file, buffer);
-    }
-
-    printf("\rWriting numbers (%d/%d)... OK\n", 10, 10);
-
-    // create a separate function for this as well maybe?
-    printf("Seeking file... ");
-    fflush(stdout);
-    status = fseek(numbers_file, 0, SEEK_SET);
-    printf("%s\n", (status < 0 ? "Fail :(" : "OK"));
-    if (status < 0) {
-        error("error: %s (%d)\n", strerror(errno), -errno);
-    }
-
-    long pos = 0;
-
-    // Go through and increment the numbers
-    for (int i = 0; i < 10; i++) {
-        printf("\rIncrementing numbers (%d/%d)... ", i, 10);
-        fflush(stdout);
-
-        // Get current stream position
-        pos = ftell(numbers_file);
-
-        // Parse out the number and increment
-        int32_t number;
-        fscanf(numbers_file, "%ld", &number);
-        number += 1;
-
-        // Seek to beginning of number
-        fseek(numbers_file, pos, SEEK_SET);
-
-        // Store number
-        sprintf(buffer, "    %d\n", number);
-        FileWrite(numbers_file, buffer);
-
-        // Flush between write and read on same file
-        fflush(numbers_file);
+    FILE* data_file = FileOpen("/fs/data.csv");
+
+    // write the first row to the file with some arbitrary sensor names
+    FileWrite(data_file, "Time (sec), LinPot1 (in/s), LinPot2 (in/s), LinPot3 (in/s), LinPot4 (in/s)\n");
+
+    // fill the file with arbitrary numbers (this is just a proof of concept, these are intentionally bs)
+    clock_t timer = clock();
```

Comment:
Optional: Simplify this to an incrementing integer
I will scrap it either way soon